### PR TITLE
Autreg 107 entity html view

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: api/src/main/resources/entitytemplate.hbs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>{{id}}</title>
+</head>
+<body>
+<h1>{{id}}</h1>
+<script type="text/javascript">
+let bodyJson = JSON.parse({{{body}}})
+var listDiv = document.getElementById('entity-body');
```
[the remark this answers is not in the document]
Comment (brinxmat) [reply api/src/main/resources/entitytemplate.hbs]: Using data-automation-id, we can easily access the elements in the rendered html when using Cypress.

File: cypress/support/step_definitions/anonymous_user_view_entity_HTML.js [hunk 2]
```diff
@@ -24,6 +24,8 @@ when(/the anonymous user requests the entity specifying an Accept header with va
 
 then('anonymous user can view the data in the given format', () => {
 	cy.get('@htmlResponse').then((response) => {
-		assert.contains(response.body, "html-kode");
+		expect(response.body).to.have.string('<html>')
+		expect(response.body).to.have.string('<body>')
+		expect(response.body).to.have.string('<li data-automation-id="preferredLabel">')
```
Comment (brinxmat) [inline cypress/support/step_definitions/anonymous_user_view_entity_HTML.js]: I think I'd prefer tests like cy.get('[data-automation-id=preferredLabel]').should('have.text', 'blah')

File: cypress/support/commands.js [hunk 3]
```diff
@@ -120,31 +120,19 @@ function createEntity(registryName, apiKey, dataFile) {
 
 function deleteRegistry(registryName, apiKey){
 	cy.log('deleting registry...')
-	
+
 	cy.log('api-key = ' + apiKey)
-	let emptyUrl = '/registry/' + registryName + '/empty'
+
+	let url = '/registry/' + registryName
```
Comment (brinxmat) [inline cypress/support/commands.js]: Do we modify url?

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            Template template = handlebars.compile(ENTITYTEMPLATE);
+            writer.write(template.apply(entityMap));
+
+            writer.flush();
+        }
+    }
+
+    private String findTitle(List<?> prefferedLabel) {
+        String label = NO_LABEL;
+        if(!Objects.isNull(prefferedLabel)) {
```
Comment (brinxmat) [inline api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: nonNull

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    private static final String NO_LABEL = "(No label)";
+    private static final String VALUE = "value";
+    private static final String LANG = "lang";
+    private static final String LANG_NO = "no";
+    private static final String LANG_EN = "en";
+    private static final String PREFERRED_LABEL = "preferredLabel";
+    private static final String ID = "id";
+    private static final String BODY = "body";
+    private static final String ENTITYTEMPLATE = "entitytemplate";
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: fixed

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+            Handlebars handlebars = new Handlebars();
+            Template template = handlebars.compile(ENTITYTEMPLATE);
+            writer.write(template.apply(entityMap));
+
+            writer.flush();
+        }
+    }
+
+    private String findTitle(List<?> prefferedLabel) {
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: fixed

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            @SuppressWarnings("unchecked")
+            Map<String, String> titleMap = prefferedLabel.stream().filter(labelObject -> 
+            ((Map<String, String>)labelObject).get(LANG).equals(LANG_EN)||
+            ((Map<String, String>)labelObject).get(LANG).equals(LANG_NO))
+            .collect(Collectors.toMap(
+                    labelObject -> ((Map<String, String>)labelObject).get(LANG), 
+                    labelObject -> ((Map<String,String>)labelObject).get(VALUE)));
+            if(!Objects.isNull(titleMap.get(LANG_NO))) {
+                label = titleMap.get(LANG_NO);
+            } else if(!Objects.isNull(titleMap.get(LANG_EN))) {
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: fixed

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    }
+
+    private String findTitle(List<?> prefferedLabel) {
+        String label = NO_LABEL;
+        if(!Objects.isNull(prefferedLabel)) {
+            @SuppressWarnings("unchecked")
+            Map<String, String> titleMap = prefferedLabel.stream().filter(labelObject -> 
+            ((Map<String, String>)labelObject).get(LANG).equals(LANG_EN)||
+            ((Map<String, String>)labelObject).get(LANG).equals(LANG_NO))
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: yes

File: cypress/support/step_definitions/anonymous_user_view_entity_HTML.js (hunk 2 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/anonymous_user_view_entity_HTML.js]: Will add a task to fix this across all tests

File: cypress/support/commands.js (hunk 3 above)
Comment (Eiriknil) [reply cypress/support/commands.js]: Changing to const, needs to be done across all tests

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            .collect(Collectors.toMap(
+                    labelObject -> ((Map<String, String>)labelObject).get(LANG), 
+                    labelObject -> ((Map<String,String>)labelObject).get(VALUE)));
+            if(!Objects.isNull(titleMap.get(LANG_NO))) {
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: fixed

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java (hunk 4 above)
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: fixed

File: api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    private static final String VALUE = "value";
+    private static final String LANG = "lang";
+    private static final String LANG_NO = "no";
+    private static final String LANG_EN = "en";
+    private static final String PREFERRED_LABEL = "preferredLabel";
+    private static final String ID = "id";
+    private static final String BODY = "body";
+    private static final String ENTITYTEMPLATE = "entitytemplate";
+    private static final String LABEL = "label";
+
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/model/EntityMessageBodyWriter.java]: Probably, won't fix now

**Pull request merged**